### PR TITLE
pugixml: update to 1.12.1

### PR DESCRIPTION
**Pull request by neheb**
Fixes compilation with C++17.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by eli-schwartz** [comment]
Commit message is wrong, it doesn't affect C++17 at all, here is the diff for that release: https://github.com/zeux/pugixml/compare/v1.12...v1.12.1

File: subprojects/packagefiles/pugixml/meson.build [hunk 1]
```diff
@@ -1,4 +1,4 @@
-project('pugixml', 'cpp', version: '1.12', meson_version: '>=0.46.0')
+project('pugixml', 'cpp', default_options : 'cpp_std=c++14', version: '1.12.1', meson_version: '>=0.46.0')
```
Comment (neheb) [inline subprojects/packagefiles/pugixml/meson.build:1]: this isn't working...

**Comment by xclaesse** [comment]
They did not remove usage of random_shuffle in tests.

File: subprojects/packagefiles/pugixml/meson.build (hunk 1 above)
Comment (xclaesse) [reply subprojects/packagefiles/pugixml/meson.build:1]: That should work with meson master, but with meson 0.62 that option is ignored on subprojects.

**Comment by neheb** [comment]
right. But this check: https://github.com/zeux/pugixml/commit/1312a46d91623d7a970f8248b0655e7ac28a8463#diff-395b06d00568647a628eb592af5c31b03e54648b8e7cd11ef60c359fa8b5419cR14
is not working.

File: subprojects/packagefiles/pugixml/meson.build (hunk 1 above)
Comment (neheb) [reply subprojects/packagefiles/pugixml/meson.build:1]: AFAIK meson 0.62.1 is used with CI:

```
2022-04-27T21:04:42.8479195Z Collecting meson
2022-04-27T21:04:42.8952087Z   Downloading meson-0.62.1-py3-none-any.whl (863 kB)
2022-04-27T21:04:43.0333993Z Installing collected packages: meson
2022-04-27T21:04:44.3168434Z Successfully installed meson-0.62.1
```

**Comment by xclaesse** [comment]
Probably https://github.com/zeux/pugixml/commit/c60ca94cdd836aac8bcee93b01f541bab24de7a5

**Comment by eli-schwartz** [comment]
More specifically, it is https://github.com/mesonbuild/meson/pull/10323

I thought we fixed this long ago, but it turns out we did not -- it got reverted because mysteriously breaking the CI apparently.

**Comment by neheb** [comment]
Fixed.

**Pull request merged**